packages/logger/src/index.ts

```diff
@@ -0,0 +1,216 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+/**
+ * @packageDocumentation
+ *
+ * A logger for fuel based on the [debug](https://www.npmjs.com/package/debug) module.
+ *
+ * @example
+ *
+ * ```TypeScript
+ * import { logger } from '@fuel-ts/logger'
+ *
+ * const log = logger('fuel-ts:my:package:name')
+ *
+ * try {
+ *   // an operation
+ *   log('something happened: %s', 'it was ok')
+ * } catch (err) {
+ *   log.error('something bad happened: %o', err)
+ * }
+ *
+ * log('with this walletAddress: %w', address)
+ * log('and this B256: %b', address)
+ * log('and this bech32 address : %n', address)
+ * ```
+ *
+ * ```console
+ * $ DEBUG=fuel-ts:* node index.js
+ * something happened: it was ok
+ * something bad happened: <stack trace>
+ * with this walletAddress: 0xf212....3aDc
+ * ```
+ */
```

Comment:
I suppose, but this is an internal tool for our own debugging purposes, so it would be documented in a similar fashion to [`@fuel-ts/utils`](https://docs.fuel.network/docs/fuels-ts/utils/) which isn't very user friendly.